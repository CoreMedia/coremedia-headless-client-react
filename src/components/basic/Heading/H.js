// @flow
import { Box } from '../Box';

const H = Box.extend`
  font-family: ${props => props.theme.typography.fontFamily.heading};
  font-weight: ${props => props.theme.typography.fontWeight.normal};
  line-height: 1.2;
  margin: 0;
  text-transform: uppercase;
`;

H.displayName = 'H';

H.defaultProps = {
  theme: {
    typography: {
      fontFamily: {
        heading: 'Helvetica',
      },
      fontWeight: {
        normal: 400,
      },
    },
  },
};

export default H;
