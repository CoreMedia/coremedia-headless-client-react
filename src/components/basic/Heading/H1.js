// @flow
import H from './H';
import { media } from '../../../styles/themes/utils';

const H1 = H.extend`
  font-size: 1.4em;
  ${props => media(props.theme.breakpoints.tablet)`font-size: 1.7em;`};
  ${props => media(props.theme.breakpoints.desktop)`font-size: 2em;`};
  ${props => media(props.theme.breakpoints.large)`font-size: 2.4em;`};
`;

H1.displayName = 'H1';

H1.defaultProps = {
  theme: {
    breakpoints: {
      tablet: 768,
      desktop: 992,
      large: 1280,
    },
    typography: {
      fontFamily: {
        heading: 'Helvetica',
      },
      fontWeight: {
        normal: 400,
      },
    },
  },
};

export default H1;
