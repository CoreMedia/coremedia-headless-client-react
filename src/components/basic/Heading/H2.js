// @flow
import H from './H';
import { media } from '../../../styles/themes/utils';

const H2 = H.extend`
  font-size: 1.3em;
  ${props => media(props.theme.breakpoints.tablet)`font-size: 1.5em;`};
  ${props => media(props.theme.breakpoints.desktop)`font-size: 1.7em;`};
  ${props => media(props.theme.breakpoints.large)`font-size: 2em;`};
`;

H2.displayName = 'H2';

H2.defaultProps = {
  theme: {
    breakpoints: {
      tablet: 768,
      desktop: 992,
      large: 1280,
    },
    typography: {
      fontFamily: {
        heading: 'Helvetica',
      },
      fontWeight: {
        normal: 400,
      },
    },
  },
};

export default H2;
