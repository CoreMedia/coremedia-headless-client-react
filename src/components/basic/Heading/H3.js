// @flow
import H from './H';
import { media } from '../../../styles/themes/utils';

const H3 = H.extend`
  font-size: 1.2em;
  ${props => media(props.theme.breakpoints.tablet)`font-size: 1.3em;`};
  ${props => media(props.theme.breakpoints.desktop)`font-size: 1.4em;`};
  ${props => media(props.theme.breakpoints.large)`font-size: 1.6em;`};
`;

H3.displayName = 'H3';

H3.defaultProps = {
  theme: {
    breakpoints: {
      tablet: 768,
      desktop: 992,
      large: 1280,
    },
    typography: {
      fontFamily: {
        heading: 'Helvetica',
      },
      fontWeight: {
        normal: 400,
      },
    },
  },
};

export default H3;
