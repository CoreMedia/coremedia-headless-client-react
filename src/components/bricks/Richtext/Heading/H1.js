// @flow
import { H1 as BasicH1 } from '../../../basic/Heading';

const H1 = BasicH1.extend`
  margin-top: 1em;
`;
H1.displayName = 'H1';

export default H1;
