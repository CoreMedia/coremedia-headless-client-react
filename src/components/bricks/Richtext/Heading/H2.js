// @flow
import { H2 as BasicH2 } from '../../../basic/Heading';

const H2 = BasicH2.extend`
  margin-top: 0.75em;
`;
H2.displayName = 'H2';

export default H2;
