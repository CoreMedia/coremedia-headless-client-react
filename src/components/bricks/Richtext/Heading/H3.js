// @flow
import { H3 as BasicH3 } from '../../../basic/Heading';

const H3 = BasicH3.extend`
  margin-top: 0.5em;
`;
H3.displayName = 'H3';

export default H3;
