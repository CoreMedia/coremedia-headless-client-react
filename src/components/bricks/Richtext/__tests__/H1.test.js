// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { H1 } from '../Heading';

describe('H1 Component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<H1>Content</H1>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
