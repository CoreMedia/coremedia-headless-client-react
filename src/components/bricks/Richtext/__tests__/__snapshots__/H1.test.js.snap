// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`H1 Component should render correctly 1`] = `
.c0 {
  margin: 0;
  padding: 0;
  line-height: 1.4;
  box-sizing: border-box;
  font-family: Helvetica;
  font-weight: 400;
  line-height: 1.2;
  margin: 0;
  text-transform: uppercase;
  font-size: 1.4em;
  margin-top: 1em;
}

@media (min-width:48em) {
  .c0 {
    font-size: 1.7em;
  }
}

@media (min-width:62em) {
  .c0 {
    font-size: 2em;
  }
}

@media (min-width:80em) {
  .c0 {
    font-size: 2.4em;
  }
}

<div
  className="c0"
>
  Content
</div>
`;
